src/ucs/config/types.h

```diff
@@ -111,6 +111,18 @@ typedef enum {
 
 typedef UCS_CONFIG_STRING_ARRAY_FIELD(names) ucs_config_names_array_t;
 
+typedef enum {
+    UCS_CONFIG_ALLOW_LIST_ALL    = UCS_BIT(0),
+    UCS_CONFIG_ALLOW_LIST_ALLOW  = UCS_BIT(1),
+    UCS_CONFIG_ALLOW_LIST_NEGATE = UCS_BIT(2)
```

Comment:
these options are mutually exclusive. I'd define just one of them, for instance `UCS_CONFIG_ALLOW_LIST_ALLOW`. Then is this flag is not set, it is negate list

Reply:
@brminich ,
From a logical perspective, there are 3 possible, **mutually exclusive** modes:
allow, allow-all, negate.

Obviously 3 bits are not "needed" to represent it and 2 are enough, but it is rather a matter of convenience.
In current implementation we can write:

```
if (allow_list->mode == UCS_CONFIG_ALLOW_LIST_NEGATE) { ...
```

According to what you suggest, it will be harder to check:
```
if (!(allow_list->mode & UCS_CONFIG_ALLOW_LIST_ALLOW) &&
    !(allow_list->mode & UCS_CONFIG_ALLOW_LIST_ALLOW_ALL)) { ...
```
(Or any other logically similar condition - it doesn't matter, the point is that 2 conditions will be needed).
Of course we can "wrap it in macro/function", but even this:
```
if (UCS_CONFIG_ALLOW_LIST_IS_NEGATE_MODE(allow_list->mode)) { ...
```
IMO looks worse than the current variant.

Are you sure that's what we want?

Reply:
I think it is more consistent with other options in the code.
@yosefe, wdyt?

Reply:
let's use regular enum values, not bits

Reply:
Done